bin/migrate-oats-data/noi/notice_of_intent_submissions/statuses/notice_of_intent_status_in_progress.py

```diff
@@ -0,0 +1,121 @@
+        ) as sql_file:
+            count_query = sql_file.read()
+            cursor.execute(count_query)
+            count_total = dict(cursor.fetchone())["count"]
+        logger.info(f"Total Notice of Intents data to update: {count_total}")
+
+        failed_inserts = 0
+        successful_updates_count = 0
+        last_application_id = 0
+
+        with open(
+            "noi/sql/notice_of_intent_submission/statuses/in_progress/notice_of_intent_status_in_progress.sql",
+            "r",
+            encoding="utf-8",
+        ) as sql_file:
+            application_sql = sql_file.read()
+            while True:
+                cursor.execute(
+                    f"{application_sql} WHERE oats_in_prog.alr_application_id > {last_application_id} ORDER BY oats_in_prog.alr_application_id;"
+                )
+
+                rows = cursor.fetchmany(batch_size)
+
+                if not rows:
+                    break
+                try:
+                    records_to_be_updated_count = len(rows)
+
+                    _update_records(conn, batch_size, cursor, rows)
+
+                    successful_updates_count = (
+                        successful_updates_count + records_to_be_updated_count
+                    )
+                    last_application_id = dict(rows[-1])["alr_application_id"]
+
+                    logger.debug(
+                        f"retrieved/updated items count: {records_to_be_updated_count}; total successfully updated notice of intents so far {successful_updates_count}; last updated alr_application_id: {last_application_id}"
+                    )
+                except Exception as err:
+                    # this is NOT going to be caused by actual data update failure. This code is only executed when the code error appears or connection to DB is lost
+                    logger.exception()
+                    conn.rollback()
+                    failed_inserts = count_total - successful_updates_count
+                    last_application_id = last_application_id + 1
+
+    logger.info(
+        f"Finished {etl_name}: total amount of successful updates {successful_updates_count}, total failed updates {failed_inserts}"
+    )
+
+
+def _update_records(conn, batch_size, cursor, rows):
+    parsed_data_list = _prepare_oats_data(rows)
+
+    if len(parsed_data_list) > 0:
+        execute_batch(
+            cursor,
+            _update_query,
+            parsed_data_list,
+            page_size=batch_size,
+        )
+
+    conn.commit()
+
+
+_update_query = """
+                    UPDATE alcs.notice_of_intent_submission_to_submission_status 
+                            SET effective_date  = %(date)s
+                    WHERE submission_uuid = %(uuid)s and status_type_code = 'PROG'
+"""
+
+
+def _prepare_oats_data(row_data_list):
+    data_list = []
+    for row in row_data_list:
+        data = map_fields(dict(row))
+        data_list.append(data)
+    return data_list
+
+
+def map_fields(data):
```

Comment:
question: Do some start with _ to indicate they are private?

Reply:
exactly! and a good catch